### PR TITLE
[ACS-3968] Layout changes for Workspace Side Menu

### PR DESCRIPTION
**Pull request by Yasa-Nataliya**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [x ] Code style update (formatting, local variables)
> - [x ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**
Change the side navigation according to the ACS-3968 jira ticket.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
[[ACS-3968]](https://alfresco.atlassian.net/browse/ACS-3968)

[ACS-3968]: https://alfresco.atlassian.net/browse/ACS-3968?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ACS-3968]: https://alfresco.atlassian.net/browse/ACS-3968?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright © 2005 - 2021 Alfresco Software, Ltd. All rights reserved.
+ *
+ * License rights for this program may be obtained from Alfresco Software, Ltd.
+ * pursuant to a written agreement and any use of this program without such an
+ * agreement is prohibited.
+ */
+
+import { Component, OnInit } from '@angular/core';
+import { ContentApiService } from '../../../../../../../projects/aca-shared/src/public-api';
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.ts:10]: incorrect path

File: projects/aca-content/src/lib/components/sidenav/components/expand-menu.component.html [hunk 2]
```diff
@@ -10,7 +10,6 @@
       mat-button
       class="action-button full-width"
     >
-      <adf-icon *ngIf="item.icon" [value]="item.icon"></adf-icon>
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/components/expand-menu.component.html:13]: why icon is removed from the menu component? is that by design to not have any icons?

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: projects/aca-content/src/lib/components/layout/app-layout/app-layout.component.html [hunk 3]
```diff
@@ -1,21 +1,14 @@
...
+    [sidenavMin]="0"
+    [sidenavMax]="289"
     [stepOver]="600"
     [hideSidenav]="hideSidenav"
     [expandedSidenav]="expandedSidenav"
     (expanded)="onExpanded($event)"
   >
-    <adf-sidenav-layout-header>
-      <ng-template let-isMenuMinimized="isMenuMinimized">
-        <app-header role="heading" aria-level="1" *ngIf="!hideSidenav" (toggleClicked)="layout.toggleMenu()" [expandedSidenav]="!isMenuMinimized()">
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/layout/app-layout/app-layout.component.html:13]: this removes the collapsible support for the sidebar, I think it should still be supported

File: projects/aca-content/src/lib/components/layout/app-layout/app-layout.component.html [hunk 4]
```diff
@@ -1,21 +1,14 @@
 <adf-upload-drag-area [rootFolderId]="currentFolderId" [disabled]="!canUpload">
   <adf-sidenav-layout
     #layout
-    [sidenavMin]="70"
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/layout/app-layout/app-layout.component.html:4]: why this change is needed?

**Review by DenysVuika — CHANGES_REQUESTED** [review]

**Comment by DenysVuika** [comment]
**Bug: the sidebar no longer collapses, the content disappears**

Screenshot 2023-01-09 at 11 08 39

**Bug: the text in the sidebar menu items is not aligned correctly, and seems to be at the bottom instead of the middle:**

image

**Bug: there's no way to create a new content, the new folder button is missing, probably other "create" items too**

**Comment by DenysVuika** [comment]
@MohinishSah please help @Yasa-Nataliya with code reviews

**Comment by DenysVuika** [comment]
Please also check the ADW linked to this branch, due to the changes in the headers I think a lot of e2e tests are going to fail or need to be adapted to your changes

File: projects/aca-content/src/lib/components/sidenav/components/expand-menu.component.html (hunk 2 above)
Comment (Yasa-Nataliya) [reply projects/aca-content/src/lib/components/sidenav/components/expand-menu.component.html:13]: Icons are not required in the design according to the requirement

File: projects/aca-content/src/lib/components/layout/app-layout/app-layout.component.html (hunk 3 above)
Comment (Yasa-Nataliya) [reply projects/aca-content/src/lib/components/layout/app-layout/app-layout.component.html:13]: No, It's not supported. In the existing layout, collapsible is shown in the header, and in the new layout, collapse functionality is shown in the side navigation

Comment (Yasa-Nataliya) [reply projects/aca-content/src/lib/components/layout/app-layout/app-layout.component.html:13]: Once the sidenav and header changes are approved, I will raise a separate PR for collapse and expand functionality.

**Comment by Yasa-Nataliya** [comment]
> **Bug: the sidebar no longer collapses, the content disappears on screen resize**
> 
> Screenshot 2023-01-09 at 11 08 39
> 
> **Bug: the text in the sidebar menu items is not aligned correctly, and seems to be at the bottom instead of the middle:**
> 
> image
> 
> **Bug: there's no way to create a new content, the new folder button is missing, probably other "create" items too**

In this PR, create and upload, and search buttons are included: https://github.com/Alfresco/alfresco-content-app/pull/2866.

File: projects/aca-content/src/lib/components/sidenav/directives/expansion-panel.directive.ts [hunk 5]
```diff
@@ -58,10 +58,10 @@ export class ExpansionPanelDirective implements OnInit, OnDestroy {
   constructor(private store: Store<any>, private router: Router, private expansionPanel: MatExpansionPanel) {}
 
   hasActiveLinks() {
-    if (this.acaExpansionPanel && this.acaExpansionPanel.children) {
+    if (this.acaExpansionPanel && !this.acaExpansionPanel.children) {
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/directives/expansion-panel.directive.ts:61]: this does not make any sense, why you are changing this function? 
as per your change, it checks if there no children, and then enumerate the missing children? this is plain stupid

File: projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+import { AlfrescoApiService } from '@alfresco/adf-core';
+import { PeopleApi } from '@alfresco/js-api';
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-navigation-menu',
+  templateUrl: './navigation-menu.component.html',
+  styleUrls: ['./navigation-menu.component.scss']
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.ts:16]: should be having `ViewEncapsulation.None` for the view encapsulation

File: projects/aca-content/src/lib/ui/variables/variables.scss [hunk 7]
```diff
@@ -73,6 +77,13 @@ $defaults: (
   --theme-about-panel-border-color: $grey-background,
   --theme-about-panel-background-color: #ffffff,
   --theme-about-panel-title-color: #212121,
+  --theme-sidenav-background-color: $sidenav-background-color,
+  --theme-selected-text-color: $selected-text-color,
+  --theme-selected-background-color: $selected-background-color,
+  --theme-hover-background-color: $grey-text-background,
+  --theme-action-button-text-color: $action-button-text-color,
+  --theme-header-border-color: $grey-background,
+  --theme-tooltip-background-color: $pagination-background-color,
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/ui/variables/variables.scss:86]: this does not make sense, each time a pagination theme changed, it will be affecting the tooltips?

File: projects/aca-folder-rules/assets/i18n/en.json [hunk 8]
```diff
@@ -3,6 +3,11 @@
     "ACTIONS": {
       "MANAGE_RULES": "Manage Rules"
     },
+    "TOOLTIPS": {
+      "COLLAPSE_NAVIGATION": "Collapse Navigation",
+      "OPTION_SETTINGS": "Option Settings",
```
Comment (DenysVuika) [inline projects/aca-folder-rules/assets/i18n/en.json:8]: Can you confirm with the design this is the expected wording? Sounds like "Options" or "Settings" might be a better option

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: projects/aca-folder-rules/assets/i18n/en.json (hunk 8 above)
Comment (Yasa-Nataliya) [reply projects/aca-folder-rules/assets/i18n/en.json]: As Shane said, it is an options and settings, so I have changed it accordingly.

**Comment by Yasa-Nataliya** [comment]
> > **Bug: the sidebar no longer collapses, the content disappears on screen resize**
> > Screenshot 2023-01-09 at 11 08 39
> > **Bug: the text in the sidebar menu items is not aligned correctly, and seems to be at the bottom instead of the middle:**
> > image
> > **Bug: there's no way to create a new content, the new folder button is missing, probably other "create" items too**
> 
> In this PR, create and upload, and search buttons are included: #2866.

The screen resize bug has been fixed, and the menu item button text is aligned now

File: projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+  encapsulation: ViewEncapsulation.None
+})
+export class NavigationMenuComponent implements OnInit {
+  peopleApi: PeopleApi;
+  personDetails;
+  userName = '';
+  userMail = '';
+
+  constructor(apiService: AlfrescoApiService) {
+    this.peopleApi = new PeopleApi(apiService.getInstance());
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.ts:26]: check other files how the api is initialized via the property getters, in your case you may have issues with logoff/logon with different accounts, or accessing the api instance before it is initialized

File: projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+@Component({
+  selector: 'app-navigation-menu',
+  templateUrl: './navigation-menu.component.html',
+  styleUrls: ['./navigation-menu.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class NavigationMenuComponent implements OnInit {
+  peopleApi: PeopleApi;
+  personDetails;
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.ts:21]: missing typing and does not make sense to have it here, as you are not using it

File: projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.ts [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+  userName = '';
+  userMail = '';
+
+  constructor(apiService: AlfrescoApiService) {
+    this.peopleApi = new PeopleApi(apiService.getInstance());
+  }
+
+  ngOnInit() {
+    this.peopleApi.getPerson('-me-').then((userInfo) => {
+      this.personDetails = userInfo?.entry;
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.ts:31]: just use local variable instead, `const personDetails = userInfo?.entry` or just access those entry properties directly without this variable

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: projects/aca-content/src/lib/components/sidenav/sidenav.component.scss [hunk 12]
```diff
@@ -1,150 +1,171 @@
...
+    .adf-container-full-width {
+      margin-left: 289px !important;
+    }
   }
 }
 
-[dir='rtl'] .sidenav {
-  .mat-expansion-panel-header {
-    padding: 0 0 0 8px !important;
+mat-tooltip-component {
```
[the remark this answers is not in the document]
Comment (DenysVuika) [reply projects/aca-content/src/lib/components/sidenav/sidenav.component.scss:162]: why do we need customizsed tooltip in this place and not for other tooltips in other places?

File: projects/aca-content/src/lib/components/layout/app-layout/app-layout.component.html (hunk 3 above)
Comment (DenysVuika) [reply projects/aca-content/src/lib/components/layout/app-layout/app-layout.component.html:13]: you can add these changes to this PR, as without that some e2e tests may be failing and PR won't get merged

File: projects/aca-content/src/lib/components/sidenav/components/expand-menu.component.html [hunk 13]
```diff
@@ -10,36 +10,30 @@
...
       <span class="action-button__label">{{ item.title | translate }}</span>
     </button>
   </div>
 </ng-container>
 
 <ng-container *ngIf="item.children && item.children.length">
   <mat-expansion-panel
-    [expanded]="acaExpansionPanel.hasActiveLinks()"
+    [expanded]="(isRouteAbout() || isRouteProfile() || isRouteSearch()) ? !acaExpansionPanel.hasActiveLinks() : acaExpansionPanel.hasActiveLinks()"
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/components/expand-menu.component.html:20]: I was saying this hundred of times already, please create 1 function in the component that evaluates the state, don't move the code to the HTML as you are not then testing it properly. 

`[expanded]="isItemExpanded(item)`

also, you make every panel expanded for all the children, what are you trying to achieve here?

**Comment by DenysVuika** [comment]
needs rebase

**Review by DenysVuika — APPROVED** [review]
needs to have the green e2e, missing "Create" button back, and some fixes to the user name layout (see jira)

File: projects/aca-content/src/lib/components/header/header.component.html [hunk 14]
```diff
@@ -1,19 +0,0 @@
-<adf-layout-header
```
Comment (BSekula) [inline projects/aca-content/src/lib/components/header/header.component.html:1]: We should not remove header html tags here, if we do not need this component, we should remove the whole component, or remove usage of this component. 

Now we are going to have `header.component` without any UI to display
